openedx/features/colaraz_features/middlewares.py

```diff
@@ -28,21 +28,28 @@ def process_request(self, request):
         """
         if getattr(settings, 'COLARAZ_ENABLE_AUTH_MIDDLEWARE', False):
             user = request.user
-            
+
             # Blocking all the paths which need to be shown to logged in users only
             blocked_sub_paths = getattr(settings, 'COLARAZ_BLOCKED_SUB_PATHS', [])
             blocked_full_paths = getattr(settings, 'COLARAZ_BLOCKED_FULL_PATHS', [])
+            allowed_paths = getattr(settings, 'COLARAZ_ALLOWED_SUB_PATHS', [])
+
+            # Allow API calls
+            is_allowed = [allowed_path in request.path for allowed_path in allowed_paths]
+
+            # Blocking all the paths which need to be shown to logged in users only
             is_blocked = [blocked_path in request.path for blocked_path in blocked_sub_paths]
             is_blocked += [blocked_path == request.path for blocked_path in blocked_full_paths]
 
-            if not user.is_authenticated and any(is_blocked):
+            # Allowed URLS will have priority over blocked URLS.
+            if not is_allowed and not user.is_authenticated and any(is_blocked):
```

Comment:
This will not work as is_allowed will always be a non-empty list with true or false as elements.
kindly see the above comment it will solve the whole problem.

Reply:
nicee catchh (y)